datafusion/common/src/stats.rs

```diff
@@ -151,6 +151,13 @@ impl Precision<usize> {
             (_, _) => Precision::Absent,
         }
     }
+
+    /// Return the estimate of applying a filter of selectivity `selectivity` to
+    /// this Precision. A selectivity of `1.0` means that all rows are selected.
+    /// A selectivity of `0.5` means half the rows are selected.
+    pub fn apply_filter(self, selectivity: f64) -> Self {
```

Comment:
This is one of the key APIs I expect to change as part of https://github.com/apache/arrow-datafusion/issues/8078 (the output will retain information about the min/max).

Reply:
I am wondering if it might be more suitable to locate this function in physical_plan's? There's already a multiplication function here, and it seems to me that this function could potentially be more relevant with filter context.

Reply:
This function both multiplies the values **and** turns the statistics into inexact (which I actually missed my first time through this code).

I think the intent is much clearer and consistent in this PR where the logic is commented in a function (rather than replicated 4 times, inconsistently).

That being said, I agree the notion of 'filter' is probably more specific to the physical plan rather than a "precision"

Perhaps we can come up with a different name for the function ? what about `Precision::estimate_filtered` or `Precision::with_estimated_selectivity`?

Reply:
I think `Precision::with_estimated_selectivity` sounds more descriptive.

Reply:
renamed